tsdb/head_read.go

```diff
@@ -266,10 +266,10 @@ func (h *headChunkReader) Chunk(ref chunks.ChunkRef) (chunkenc.Chunk, error) {
 		return nil, storage.ErrNotFound
 	}
 
-	s.Lock()
+	s.RLock()
 	c, garbageCollect, err := s.chunk(cid, h.head.chunkDiskMapper)
```

Comment:
is this read?

Reply:
It only reads data out of HEAD, but when it does so it might allocate a new object.
I believe that the `garbageCollect` flag is only to signal if it's safe to discard the result (since it was created just to service the read) but I might be mistaken here.